### PR TITLE
Testcases

### PR DESCRIPTION
**Pull request by RajeevKumar-Singh**

### CODE REVIEW

File: src/test/java/com/techsophy/tsf/wrapperservice/CommentServiceTest.java [hunk 1]
```diff
@@ -114,4 +86,58 @@ void getComments() throws Exception
...
+        TaskModel task = Mockito.mock(TaskModel.class);
+        CommentDTO commentDTO = new CommentDTO(null, null, "bKey", "comment");
+        PaginationDTO<List<TaskInstanceDTO>> page = new PaginationDTO<>(List.of(taskInstanceDTO), 1, 1, 1, 1l, 1l);
+        Mockito.when(restTemplate.exchange(anyString(), any(HttpMethod.class), any(), eq(Object.class))).thenReturn(response);
+        Mockito.when(response.getStatusCode()).thenReturn(HttpStatus.BAD_REQUEST);
+        Mockito.when(processService.getTasksByQuery(any(), any(), any())).thenReturn(page);
+        Mockito.when(objectMapper.convertValue(any(), any(TypeReference.class))).thenReturn(List.of(task));
+
+        Map<String, Object> actualOutput = commentService.createComment(commentDTO);
+        Map<String, Object> expectedOutput = objectMapper.convertValue(response, Map.class);
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/wrapperservice/CommentServiceTest.java:125]: Mockito.when(response.getStatusCode()).thenReturn(HttpStatus.BAD_REQUEST); 
Map<String, Object> expectedOutput = objectMapper.convertValue(response, Map.class);

should be on line 115 to make it more readable.

File: src/test/java/com/techsophy/tsf/wrapperservice/config/JWTRoleConverterTest.java [hunk 2]
```diff
@@ -68,4 +69,19 @@ void convertTestWhileThrowingException() throws JsonProcessingException {
...
     }
+
+    @Test
+    void convertTestWhileKeyAsClientRoles() throws JsonProcessingException {
+        Jwt jwt = Mockito.mock(Jwt.class);
+        List<String> awgmentRolesList = new ArrayList<>();
+        String userResponce = "userResponse";
+        Mockito.when(webClientWrapper.webclientRequest(any(), any(), any(), any())).thenReturn(userResponce);
+        Mockito.when(objectMapper.readValue(userResponce, Map.class)).thenReturn(Map.of("key", "val"));
+        Mockito.when(objectMapper.readValue(userResponce,Map.class)).thenReturn(Map.of(CLIENT_ROLES, "val"));
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/wrapperservice/config/JWTRoleConverterTest.java:80]: duplicate line?

Comment (VaibhavTechsophy) [reply src/test/java/com/techsophy/tsf/wrapperservice/config/JWTRoleConverterTest.java:80]: where is CLIENT_ROLES used in verification?

File: src/test/java/com/techsophy/tsf/wrapperservice/service/impl/AllTasksServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    }
+
+    @Test
+    void allTasksCountTest() throws JsonProcessingException {
+        Mockito.when(webClientWrapper.createWebClient(tokenUtils.getTokenFromContext())).thenReturn(webClient);
+        String response = "response";
+        Mockito.when(webClientWrapper.webclientRequest(any(WebClient.class), anyString(), anyString(), any(AllTasksCountDTO.class))).thenReturn(response);
+
+        TaskCountDTO actualOutput = allTasksService.allTasksCount(allTasksCountDTO);
+        TaskCountDTO expectedOutput = objectMapper.readValue(response,TaskCountDTO.class);
```
Comment (VaibhavTechsophy) [inline src/test/java/com/techsophy/tsf/wrapperservice/service/impl/AllTasksServiceImplTest.java:62]: objectMapper is mocked what  expected output are you getting?

Comment (VaibhavTechsophy) [reply src/test/java/com/techsophy/tsf/wrapperservice/service/impl/AllTasksServiceImplTest.java:62]: same for all instances

**Review by VaibhavTechsophy — CHANGES_REQUESTED** [review]

File: src/test/java/com/techsophy/tsf/wrapperservice/config/JWTRoleConverterTest.java (hunk 2 above)
Comment (RajeevKumar-Singh) [reply src/test/java/com/techsophy/tsf/wrapperservice/config/JWTRoleConverterTest.java:80]: In JWTRoleConvertor.java, Line No. 55 is 
if(userInformationMap.containsKey(CLIENT_ROLES))

It is checking if the map contains any Key as CLIENT_ROLES, To Cover this If-Block, I returned a map which contains CLIENT_ROLES as a key.

Comment (VaibhavTechsophy) [reply src/test/java/com/techsophy/tsf/wrapperservice/config/JWTRoleConverterTest.java:80]: the userResponse(fix the spelling) variable should be initialized with client roles, and real method should be called from mock

**Pull request closed without merging**